scripts/lcov.sh

```diff
@@ -1,10 +1,52 @@
 #!/bin/sh
 
-	rm -rf Coverage
-	lcov --capture --initial --directory library -o files.info
-	lcov --rc lcov_branch_coverage=1 --capture --directory library -o tests.info
-	lcov --rc lcov_branch_coverage=1 --add-tracefile files.info --add-tracefile tests.info -o all.info
-	lcov --rc lcov_branch_coverage=1 --remove all.info -o final.info '*.h'
-	gendesc tests/Descriptions.txt -o descriptions
-	genhtml --title "mbed TLS" --description-file descriptions --keep-descriptions --legend --branch-coverage -o Coverage final.info
-	rm -f files.info tests.info all.info final.info descriptions
+help () {
+    cat <<EOF
+Usage: $0
+Collect coverage statistics of library code into an HTML report.
+
+General instructions:
+1. Build the library with CFLAGS="--coverage -O0 -g3".
```

Comment:
Pre-existing (in the comment in the Makefile): If I run `make CFLAGS="--coverage -O0 -g3"` I'm getting failures, starting with `fuzz_x509csr` (other programs work fine). We should either (1) adapt the Makefiles to make this work, or (2) add `LDFLAGS=--coverage` to the instructions above, which is what `basic-build-test.sh` is doing.

If we go with (1), I'd recommend removing the setting of `LDFLAGS` in `basic-build-test.sh` so that building without it is tested in the CI.

Reply:
> the covtest target in the Makefile is kind of superseded by basic-build-test.sh these days

Not exactly: maybe it's useful to run `make covtest` in a specific configuration? And `basic-build-test.sh` might one day evolve to run multiple configurations. I don't think I've ever run `make covtest` so I'm not sure it's useful to have a convenient target for it, but I don't think it hurts to have it so I'll keep it.

Reply:
Regarding `covtest`, I can say I introduced it so that it would be a standardized target for coverage testing for comparison purposes, and nowadays this role is fulfilled by `basic-build-test.sh` - in a better way as it really runs all the tests we have (including those with next/legacy version of OpenSSL/GnuTLS). So, it's no longer useful for its original purpose.

Whether it's useful for any other purpose is anyone's guess. What I can say is that I've not been using it since we have `basic-build-test.sh` and I don't see myself using it in the future. Generally speaking, either I'm interested in coverage for one specific area and I'll run just the test for that area, or in comparisons between versions, and I'll use `basic-build-test.sh`. If I'm interested in more fancy comparisons, like between configurations, or versions of `compat.sh`, etc, I'm likely to write a dedicated script for that and not really need this 4-lines target. Others might have different experiences.

> I don't think it hurts to have it so I'll keep it.

I think having too many things can hurt by creating confusion if not explained. This looks like it's all the available test except it's not really. If you keep this as it is (instead of making it just call `basic-build-test.sh`, as I would prefer) please add a comment pointing to `basic-build-test.sh` and explaining the differences briefly - and in CMakeLists.txt too.

Reply:
>  If you keep this as it is [...] please add a comment pointing to `basic-build-test.sh`

Oh, I see you already did that in the next commit, great! Can you add a similar comment to the covtest target in `CMakeLists.txt` though?

Reply:
Ok, since it does create confusion and it's not really useful, I've removed the target. I added a bit more documentation on how to do coverage. It duplicates what's in `lcov.sh` but this way it's in a place where people are likely to look (`lcov.sh` is not discoverable, and `make lcov` only a little more, but `grep coverage Makefile` is more natural).